pkg/controller/util/endpoint/controller_utils.go

```diff
@@ -32,6 +34,75 @@ import (
 // EndpointsMatch is a type of function that returns true if pod endpoints match.
 type EndpointsMatch func(*v1.Pod, *v1.Pod) bool
 
+// ServiceSelectorCache is a cache of service selectors to avoid high CPU consumption caused by frequent calls to AsSelectorPreValidated (see #73527)
+type ServiceSelectorCache struct {
+	cache map[string]labels.Selector
+	lock  sync.RWMutex
```

Comment:
nit: move the lock above the fields it is protecting

Reply:
addressed